### PR TITLE
fix!: reorganize chart and vegalite and rename vegalite component

### PR DESCRIPTION
**Pull request by lyjeileen**
## Change:
- Rename vegaLiteChart to vegaLite
- Move chart and vegaLite to Visualization folder

More storybook examples will be added in a follow-up PR

### Before
Screenshot 2024-05-22 at 4 14 03 PM

### After
Screenshot 2024-05-22 at 4 12 45 PM

### CODE REVIEW

File: src/components/visualization/vegaLite/vegaLite.stories.tsx [hunk 1]
```diff
@@ -29,9 +29,10 @@ meta.argTypes = {
   },
   options: {
     description:
-      'The options object is passed directly to the vega-lite chart. Refer to the [Vega-lite documentation](https://vega.github.io/vega-lite/) for more information.',
+      'The options object is passed directly to the vega-lite. Refer to the [Vega-lite documentation](https://vega.github.io/vega-lite/) for more information.',
```
Comment (kaseyvee) [inline src/components/visualization/vegaLite/vegaLite.stories.tsx:32]:
```suggestion
      'The options object is passed directly to Vega-Lite. Refer to the [Vega-Lite documentation](https://vega.github.io/vega-lite/) for more information.',
```

File: src/components/visualization/vegaLite/vegaLite.stories.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import type { Meta, StoryFn } from '@storybook/react'
 import React from 'react'
 
-import VegaLiteChart from './vegaLiteChart'
-const meta: Meta<React.ComponentProps<typeof VegaLiteChart>> = {
-  title: 'Rustic UI/Chart/Vega-Lite Chart',
-  component: VegaLiteChart,
+import VegaLite from './vegaLite'
+const meta: Meta<React.ComponentProps<typeof VegaLite>> = {
+  title: 'Rustic UI/Visualization/VegaLite',
```
Comment (kaseyvee) [inline src/components/visualization/vegaLite/vegaLite.stories.tsx:6]:
```suggestion
  title: 'Rustic UI/Visualization/Vega-Lite',
```

File: src/components/types.ts [hunk 3]
```diff
@@ -139,7 +139,7 @@ export interface TableFormat extends DataFormat {
 export type TableData = TableFormat & Updates<TableFormat>
 
 type VegaLiteOptions = EmbedOptions<string, Renderers>
-export interface VegaLiteChartFormat extends DataFormat {
+export interface VegaLiteVizFormat extends DataFormat {
```
Comment (Shiti) [inline src/components/types.ts:142]: Maybe this should just be `VegaLiteFormat` 
```suggestion
export interface VegaLiteFormat extends DataFormat {
```

File: src/components/types.ts [hunk 4]
```diff
@@ -149,8 +149,7 @@ export interface VegaLiteChartFormat extends DataFormat {
   options?: VegaLiteOptions
 }
 
-export type VegaLiteChartData = VegaLiteChartFormat &
-  Updates<VegaLiteChartFormat>
+export type VegaLiteVizData = VegaLiteVizFormat & Updates<VegaLiteVizFormat>
```
Comment (Shiti) [inline src/components/types.ts:152]:
```suggestion
export type VegaLiteData = VegaLiteFormat & Updates<VegaLiteFormat>
```

**Review by Shiti — APPROVED** [review]

**Pull request merged**